gprofiler/profiler_state.py

```diff
@@ -49,6 +51,13 @@ def profiling_mode(self) -> str:
     def container_names_client(self) -> Optional[ContainerNamesClient]:
         return self._container_names_client
 
+    @property
+    def pids_to_profile(self) -> Optional[List[int]]:
+        if len(self._pids_to_profile) > 0:
```

Comment:
API wise, I think it's nicer if the list is either None (if we're not in `--pids` mode) or non-None. i.e, so this class needn't check if length is > 0. If it got a list - use the list. If not - don't use.